src/components/Carousel.tsx

```diff
@@ -1,26 +1,101 @@
 import React from 'react';
 import './Carousel.scss';
+import classNames from 'classnames';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite?: boolean;
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [slide, setSlide] = React.useState(0);
+
+  const totalItems = images.length;
+  const slideItems = Math.max(Math.ceil((totalItems - frameSize) / step), 0);
+
+  const nextSlide = infinite || slide < slideItems;
+  const previousSlide = infinite || slide > 0;
+
+  const containerStyle = React.useMemo(
+    () => ({
+      width: `${itemWidth * frameSize}px`,
+    }),
+    [itemWidth, frameSize],
+  );
+
+  const listStyle = React.useMemo(() => {
+    const maxTranslateX = (totalItems - frameSize) * itemWidth;
+    const translateX = Math.min(slide * itemWidth * step, maxTranslateX);
```

Comment:
```suggestion
    const translateX = Math.min(slide * itemWidth, maxTranslateX);
```
Now it scrolls even when I just change step and do not touch the scroll buttons (if the middle of the images list is displayed at the moment - i.e. slide !== 0).

Also, it scrolls too much, at least on Prev. button click